File: programas/while.cpp
/*
título          :while.cpp
descripción     :Ejercicios sobre ciclos.
autor           :Benjamín Valdés
c++ version  	 :11
===============================================================

 ejemplo de trabajo con ciclos en c++
*/

// biblioteca para usar cin y cout
#include<iostream>

// biblioteca para usar printf
#include<stdio.h>

using namespace std;


float potencia(int x, float n){
	int i = 0;
	float pot = 1.0;
	
	if(n > 0){
		while(i < n){
			pot =  pot * x;
			i = i + 1;
		}
	}else{
		while(0 > n){
			pot = pot * 1.0/x;
			n = n + 1;
		}
	}
	
	return pot;
}

int main(){
	
	int x;
	float n;
	
	cin >> x >> n;
	// usamos printf para controlar la cantidad de decimales
	printf("%.4f", potencia(x, n));	
}

File: programas/while.py
#título          :while.py
#descripción     :Ejercicios sobre ciclos.
#autor           :Benjamín Valdés
#python_version  :3.5.2
#===============================================================

# ejemplos para ciclos en python


def potencia(x,n):

    """ culcula la potencia de un número usando ciclos"""

    i = 0
    pot = 1.0
    if(n > 0):
        while(i < n):
            pot = pot * x
            i = i + 1
    else:
        while(0 > n):
            pot = pot * 1.0/x
            n = n + 1    
    return pot



x = int(input())
n = float(input()) 
print("%.4f" % potencia(x,n))
    